lib/path.js

```diff
@@ -360,11 +379,10 @@ win32.format = function(pathObject) {
 
   var dir = pathObject.dir;
   var base = pathObject.base || '';
-  if (dir.slice(dir.length - 1, dir.length) === win32.sep) {
-    return dir + base;
-  }
-
   if (dir) {
+    if (dir[dir.length - 1] === win32.sep) {
```

Comment:
Yes, I can think of at least two test cases that this fixes. I'll add some tests tonight.